docs/source/guides/scripting/customwavecreation.rst

```diff
@@ -0,0 +1,479 @@
+.. code-block::
+
+    global function initFrontierDefenseData
+    void function initFrontierDefenseData()
+
+- These are what starts the script and calls the functions needed for the gamemode.
+
+Brackets
+########
+.. code-block::
+
+    {
+    }
+
+- This is what houses the main script from start to finish
+
+ShopPosition and Angles
+#######################
+.. code-block::
+
+    shopPosition = < -3862,1267,1060 >
+    shopAngles =  <0,30,0>
+
+- shopPosition spawns the shop at in-game coordinates via X,Y,Z
+- shopAngles spawns the shop with rotations if defined via Y,A,P (Yaw, Angle, Pitch)
+
+SpawnPosition and Angles
+########################
+.. code-block::
+
+    FD_spawnPosition = < -838,2629,1592 >
+    FD_spawnAngles = < 0,180,0 >
+
+- FD_spawnPosition and FD_spawnAngles are what tell the game to start the Dropship animation, keep in mind this can be a bit finicky when it comes to positioning it as when respawning, the ship may clip through brushes and terrain.
+- For a better explanation of this, look in `Spawning Types(For Pilots)`_
+.. note ::
+    Without spawnPosition and spawnAngles, the game **will crash!**
+
+int index
+#########
+
+.. code-block::
+    
+    int index = 1
+
+- This tell the gamescript (Squirrel) to start the wave script.
+
+
+waveEvents
+##########
+
+.. code-block::
+
+    array<WaveEvent> wave1
+    waveEvents.append(wave1)
+
+- This is the heart of the script, what tells the game where to spawn/do in-game Events.
+
+Appending Wave Events
+#####################
+
+.. code-block::
+
+    array<WaveEvent> wave1
+    wave1.append(CreateSmokeEvent(< 4060,771,1004 >,90,index++))
+    wave1.append(CreateWaitForTimeEvent(5,index++))
+    wave1.append(CreateWarningEvent(FD_IncomingWarnings.ArcTitanAlt,index++))
+    wave1.append(CreateArcTitanEvent(< 4300,771,1004 >,< 0,180,0 >,"",index++))
+    wave1.append(CreateDroppodGruntEvent(< 3733,100,1004 >,"",0))
+    waveEvents.append(wave1)
+
+
+- Now this is where it gets interesting, inside of the array, this is where you put your Events at for each wave. 
+- There are multiple types of events that you can use (Enemies, Time, Smoke, Etc.) which to script it, it always starts with ``wave#.append`` Then the Event type ``CreateSmokeEvent``, then the variables that come after it ``(<0,0,0>,90,index++))`` (Which varies on events, which will be Referenced in `Cheat Sheet`_).
+- Always, the formula for writing an append goes as follows ``wave#.append(EventHere(VariablesHere,index++/0))``
+- An example of this is as follows ``wave1.append(CreateSmokeEvent(<0,0,0>,90,index++))``
+- ``index++`` or ``0`` is very important to include at the end of your variable listing as shown in the example above, if you do not include it, your script **will crash**!
+    - The difference between the two is, ``index++`` advances the script one line in the array as ``0`` is used on the very last wave append to end the array. Failure to include ``0`` at the end of an array **will crash**!
```

Comment:
not really a good explanation, having all the ``CreateX`` would also be good
this isnt a power point you dont need to make everything in short sentences